src/test/java/com/openlattice/rehearsal/search/SearchControllerTest.kt

```diff
@@ -373,12 +411,104 @@ class SearchControllerTest : MultipleAuthenticatedUsersBase() {
+        Thread.sleep(1000)
+
+        val searchMax0 = SearchTerm("*", 0, 0)
+        var searchedEntities = searchApi.executeEntitySetDataQuery(es.id, searchMax0)
+
+        // should yield 0 hits and 0 totalHits
+        Assert.assertEquals(0, searchedEntities.hits.size)
+        Assert.assertEquals(numberOfEntities, searchedEntities.numHits.toInt())
+
+        val searchMax1 = SearchTerm("*", 0, 1)
+        searchedEntities = searchApi.executeEntitySetDataQuery(es.id, searchMax1)
+
+        // should yield 1 hit, but 10 totalHits
+        Assert.assertEquals(1, searchedEntities.hits.size)
+        Assert.assertEquals(numberOfEntities, searchedEntities.numHits.toInt())
+
+        val searchMax5 = SearchTerm("*", 0, 5)
+        searchedEntities = searchApi.executeEntitySetDataQuery(es.id, searchMax5)
+
+        // should yield 5 hits, but 10 totalHits
+        Assert.assertEquals(5, searchedEntities.hits.size)
+        Assert.assertEquals(numberOfEntities, searchedEntities.numHits.toInt())
+
+        val searchMax10 = SearchTerm("*", 0, 10)
+        searchedEntities = searchApi.executeEntitySetDataQuery(es.id, searchMax10)
+
+        // should yield 10 hits and 10 totalHits
+        Assert.assertEquals(numberOfEntities, searchedEntities.hits.size)
+        Assert.assertEquals(numberOfEntities, searchedEntities.numHits.toInt())
+
+        val searchFrom2Max5 = SearchTerm("*", 2, 5)
+        searchedEntities = searchApi.executeEntitySetDataQuery(es.id, searchFrom2Max5)
+
+        // should yield 4 hits and 10 totalHits
+        Assert.assertEquals(4, searchedEntities.hits.size)
+        Assert.assertEquals(numberOfEntities, searchedEntities.numHits.toInt())
+    }
+
+    @Test
+    fun testMaxHitsSearchAfterDelete() {
+        // create 100 entities
+        val et = createEntityType()
+        val es = createEntitySet(et)
+
+        val numberOfEntities = 100
+        val testData = TestDataFactory.randomStringEntityData(numberOfEntities, et.properties).values.toList()
+        val entities = dataApi.createEntities(es.id, testData).toSet().zip(testData).toMap()
+        Thread.sleep(1000)
+
+        // should be indexed automatically
+        val idsList = entities.keys.toList()
+        val searchAll = SearchTerm("*", 0, numberOfEntities)
+        var searchedEntities = searchApi.executeEntitySetDataQuery(es.id, searchAll)
+
+        Assert.assertEquals(numberOfEntities, searchedEntities.numHits.toInt())
+        searchedEntities.hits.forEach { entityData ->
+            val id = UUID.fromString(entityData.getValue(EdmConstants.ID_FQN).first() as String)
+            Assert.assertTrue(entities.keys.contains(id))
+        }
+
+        // Delete multiple entities
+        val deletedIds = (0 until 50).map {
+            idsList[Random.nextInt(numberOfEntities)]
+        }.toSet()
+        val numberOfDeletes = deletedIds.size // could pick same id
+
+        dataApi.deleteEntities(es.id, deletedIds.toSet(), DeleteType.values()[Random.nextInt(2)])
```

Comment:
deletedIds is already a set, so toSet is redundant.

re: DeleteType, I would say either pick one or do some with each delete type. If there's ever a bug with only soft types, for example, the test would start failing 50% of the time. So you could split up the IDs into two batches and delete some hard and some soft.

Reply:
Good idea! The behaviour here should be the same for both delete types.